app/controllers/admin/elections_controller.rb

```diff
@@ -1,72 +1,75 @@
 # encoding: UTF-8
 class Admin::ElectionsController < ApplicationController
-  load_permissions_and_authorize_resource find_by: :url
+  # Quick fix because Cancancan doesn't load when we use URL as param.
+  # /d.wessman https://github.com/fsek/web/issues/134
   before_action :authorize
-  before_action :set_posts, only: [:new, :show, :edit]
+  before_action :set_election, only: [:show,:edit,:update,:destroy,:candidates,:nominations]
 
   def new
+    @election = Election.new
+    @posts = Post.order(title: :asc)
   end
-
   def show
+    @posts = Post.order(title: :asc)
   end
-
   def edit
+    @posts = Post.order(title: :asc)
   end
-
   def index
     @elections = Election.order(start: :desc)
   end
-
   def create
-    if @election.save
-      redirect_to [:admin, @election], notice: 'Valet skapades'
-    else
-      render action: :new
+    @election = Election.new(election_params)
+    respond_to do |format|
+      if @election.save
+        format.html { redirect_to [:admin,@election], notice: 'Valet skapades!' }
+      else
+        format.html { render action: "new" }
+      end
     end
   end
-
   def update
-    if @election.update(election_params)
-      redirect_to [:admin, @election], notice: 'Valet uppdaterades, gött'
-    else
-      render action: :edit
+    respond_to do |format|
+      if @election.update(election_params)
+        format.html { redirect_to admin_election_path(@election), notice: 'Valet uppdaterades, gött' }
+        format.json { head :no_content }
+      else
+        format.html { render action: 'edit' }
+        format.json { render json: @election.errors, status: :unprocessable_entity }
+      end
     end
   end
 
   def destroy
     @election.destroy
-    redirect_to admin_elections_path, notice: 'Valet raderades, hoppas att det var meningen!.'
+    respond_to do |format|
+      format.html { redirect_to elections_path,notice: 'Valet raderades, hoppas att det var meningen!.' }
+      format.json { head :no_content }
+    end
   end
 
   def nominations
     @nominations_grid = initialize_grid(@election.nominations,
-                                        name: 'g2',
-                                        enable_export_to_csv: true,
-                                        csv_file_name: 'nomineringar')
+                                        :name => 'g2',
+                                        :enable_export_to_csv => true,
+                                        :csv_file_name => 'nomineringar')
   end
-
   def candidates
-    @candidates_grid = initialize_grid(@election.candidates,
-                                       name: 'candidates',
-                                       enable_export_to_csv: true,
-                                       csv_file_name: 'candidates')
+    @candidates_grid = initialize_grid(@election.candidates,name: "candidates",enable_export_to_csv: true, csv_file_name: 'candidates')
```

Comment:
Line is too long. [135/100]<br>Space missing after comma.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.